terraform/modules/hub/files/cloud-init/prometheus.sh

```diff
@@ -164,6 +164,16 @@ function run-until-success() {
   done
 }
 
+# Make sure unattended upgrades is available and we automatically reboot at 3:00
+# as required in HUB-1057.
+run-until-success "apt-get install --yes unattended-upgrades update-notifier-common"
+cat >> /etc/apt/apt.conf.d/50unattended-upgrades <<'EOF'
+Unattended-Upgrade::Automatic-Reboot "true";
+Unattended-Upgrade::Automatic-Reboot-WithUsers "false";
+Unattended-Upgrade::Automatic-Reboot-Time "03:00";
```

Comment:
Is this too crazy?

```suggestion
Unattended-Upgrade::Automatic-Reboot-Time "03:$((10 + $RANDOM % 60))";
```

Set minutes to a random number between 10 and 59 to spread out the restarts and reduce the chances of losing Prometheus coverage.

Reply:
It still leaves the issue of upgrading prod before staging/integration. Better to bork staging and not deploy to prod than just deploying blind.

Reply:
Yeah fair enough, it looks like we'd need a Concourse job for that then

Reply:
done with https://github.com/alphagov/verify-infrastructure-config/pull/602